flink/src/main/java/org/apache/iceberg/flink/actions/MigrateAction.java

```diff
@@ -0,0 +1,374 @@
+    MetricsConfig metricsConfig = MetricsConfig.fromProperties(icebergTable.properties());
+
+    List<DataFile> files = null;
+    if (spec.isUnpartitioned()) {
+      MigrateMap migrateMap = new MigrateMap(spec, nameMapping, fileFormat, metricsConfig);
+      DataStream<PartitionAndLocation> dataStream =
+          env.fromElements(new PartitionAndLocation(hiveLocation, Maps.newHashMap()));
+      DataStream<List<DataFile>> ds = dataStream.map(migrateMap);
+      try {
+        files = Lists.newArrayList(ds.executeAndCollect("migrate table :" + icebergTable.name())).stream()
+            .flatMap(Collection::stream).collect(Collectors.toList());
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      try {
+        files = importPartitions(spec, tableSource, nameMapping, fileFormat, metricsConfig, icebergTable.name());
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    AppendFiles append = icebergTable.newAppend();
+    files.forEach(append::appendFile);
+    append.commit();
+    return files.size();
+  }
+
+  private Namespace toNamespace(Namespace namespace) {
+    if (namespace.isEmpty()) {
+      return Namespace.of(icebergDbName);
+    } else {
+      String[] namespaces = new String[baseNamespace.levels().length + 1];
+      System.arraycopy(baseNamespace.levels(), 0, namespaces, 0, baseNamespace.levels().length);
+      namespaces[baseNamespace.levels().length] = icebergDbName;
+      return Namespace.of(namespaces);
+    }
+  }
+
+  private List<DataFile> importPartitions(PartitionSpec spec, ObjectPath tableSource,
+                                          String nameMapping, FileFormat fileFormat,
+                                          MetricsConfig metricsConfig, String hiveTableName)
+      throws Exception {
+    List<CatalogPartitionSpec> partitionSpecs = flinkHiveCatalog.listPartitions(tableSource);
+    List<PartitionAndLocation> inputs = Lists.newArrayList();
+    for (CatalogPartitionSpec partitionSpec : partitionSpecs) {
+      Partition partition =
+          flinkHiveCatalog.getHivePartition(flinkHiveCatalog.getHiveTable(tableSource), partitionSpec);
+      inputs.add(
+          new PartitionAndLocation(partition.getSd().getLocation(), Maps.newHashMap(partitionSpec.getPartitionSpec())));
+    }
+
+    int size = partitionSpecs.size();
+    int parallelism = Math.min(size, maxParallelism);
+
+    DataStream<PartitionAndLocation> dataStream = env.fromCollection(inputs);
+    MigrateMap migrateMap = new MigrateMap(spec, nameMapping, fileFormat, metricsConfig);
+    DataStream<List<DataFile>> ds = dataStream.map(migrateMap).setParallelism(parallelism);
+    return Lists.newArrayList(ds.executeAndCollect("migrate table :" + hiveTableName)).stream()
+        .flatMap(Collection::stream).collect(Collectors.toList());
+  }
+
+  private static class MigrateMap implements MapFunction<PartitionAndLocation, List<DataFile>> {
```

Comment:
nit: Map -> Mapper? `Map` gives the impression of Java Collection.

Here is the Flink `DataStream` map API
```
map(MapFunction<T, R> mapper)
```